tests/integration/pilot/common/routing.go

```diff
@@ -155,13 +157,11 @@ spec:
 			opts: echo.CallOptions{
 				PortName: "http",
 				Count:    1,
-				Validator: echo.And(
-					echo.ExpectOK(),
-					echo.ValidatorFunc(
-						func(response echoclient.ParsedResponses, _ error) error {
-							return response.Check(func(_ int, response *echoclient.ParsedResponse) error {
-								return ExpectString(response.RawResponse["Istio-Custom-Header"], "user-defined-value", "request header")
-							})
+				Checker: check.And(
+					check.OK(),
+					check.Func(
```

Comment:
nit: is it possible to not have the checker interface and instead just have it be a `func(r *response.Instance) error` type alias? Then we don't need to wrap everything?